packages/core/src/metrics/MetricsService.ts

```diff
@@ -65,22 +66,27 @@ export class MetricsService {
       throw new Error(`Type is invalid: ${type}`);
     }
 
+    if (!this.publisher || !this.defaultMetrics) {
+      return Promise.resolve();
+    }
+
     const payload: MetricsPayload = {
       clientId: this.getClientId(),
       type,
       timestamp: new Date().getTime(),
       data: {}
     };
 
-    metrics.forEach(m => {
-      payload.data[m.identifier] = m.collect();
-    });
+    const metricsPromise = metrics.concat(this.defaultMetrics)
```

Comment:
80-83 is a little bit convoluted, do we prefer this format instead of the `async/await`?

Reply:
Fair enough